File: 4.26/서지훈_백준_1541_잃어버린_괄호.cpp
#include <iostream>
#include <string>

int main(){
	std::string expr;
	std::cin >> expr;
	
	int result = 0, start = 0, end = 0, idx = 0;
	
	while(idx <= expr.length()){
		if(idx == expr.length() || expr[idx] == '+'){
	        result += stoi(expr.substr(start, end-start));
	        idx += 1;
	        start = idx;
	        end = idx;
	        continue;
		}
	    else if(expr[idx] == '-'){
	        result += stoi(expr.substr(start, end-start));
	        idx += 1;
	        start = idx;
	        end = idx;
	        break;
		}
		end += 1;
		idx += 1;
	}
	
	while(idx <= expr.length()){
		if(idx == expr.length() || expr[idx] == '+' || expr[idx] == '-'){
	        result -= stoi(expr.substr(start, end-start));
	        idx += 1;
	        start = idx;
	        end = idx;
	        continue;
		}
		end += 1;
		idx += 1;
	}
	
	std::cout << result;
}

File: 4.26/서지훈_백준_1541_잃어버린_괄호.py
expr = input()
expr_len = len(expr)

result = 0  # 결과 값
start, end = 0, 0  # 숫자의 시작, 끝을 표기
idx = 0
while idx <= expr_len:
    if idx == expr_len or expr[idx] == '+':
        result += int(expr[start:end])
        idx += 1
        start, end = idx, idx
        continue

    elif expr[idx] == '-':
        result += int(expr[start:end])
        idx += 1
        start, end = idx, idx
        break

    end += 1
    idx += 1

while idx <= expr_len:
    if idx == expr_len or expr[idx] == '-' or expr[idx] == '+':
        result -= int(expr[start:end])
        idx += 1
        start, end = idx, idx
        continue
    end += 1
    idx += 1

print(result)